client/processors/render_processor.py

```diff
@@ -13,49 +16,128 @@
 from resources.assets_manager import AssetsManager
 from resources.window_resource import WindowResource
 
+if TYPE_CHECKING:
+    from typing import Callable
+
+    map_layer_by_draw_function = dict[int, list[tuple[int, Callable[[], None]]]]
+
 DEBUG_COLLIDER = True
 
 logger = logging.getLogger(__name__)
 
 
 class RenderProcessor(Processor):
+    __default_camera = pyray.Camera2D((0, 0), (0, 0), 0, 1)
+
     def __init__(self):
         super().__init__()
 
     def process(self, r: ResourceManager) -> None:
         window = r.get_resource(WindowResource)
-        cameras = esper.get_components(Camera2D, Position)
         assets_manager = r.get_resource(AssetsManager)
-        if len(cameras) < 1:
-            return
-        main_camera_obj, main_camera_pos = cameras[0][1]
         pyray.clear_background(window.background_color)
-        pyray.begin_mode_2d(
-            pyray.Camera2D(
-                (main_camera_obj.offset.x, main_camera_obj.offset.y),
-                (main_camera_pos.x, main_camera_pos.y),
-                main_camera_obj.rotation,
-                main_camera_obj.zoom,
+        to_draw: map_layer_by_draw_function = {}
+        RenderProcessor.__register_textures_to_draw(assets_manager, to_draw)
+        RenderProcessor.__register_text_to_draw(to_draw, r)
+        if (
+            DEBUG_COLLIDER
+        ):  # This should be temporary until we find a better way to debug colliders
+            for ent, (pos, collider) in esper.get_components(Position, BoxCollider):
+                to_draw.setdefault(100, []).append(
+                    (0, RenderProcessor.__create_draw_rectangle_lines_ex(pos, collider))
+                )
```

Comment:
This is a good occasion to use a `defaultdict`

`from collections import defaultdict`

```suggestion
        to_draw = defaultdict(list)

        RenderProcessor.__register_textures_to_draw(assets_manager, to_draw)
        RenderProcessor.__register_text_to_draw(to_draw, r)
        
        # This should be temporary until we find a better way to debug colliders
        if DEBUG_COLLIDER:
            for ent, (pos, collider) in esper.get_components(Position, BoxCollider):
                to_draw[100].append(
                    (0, RenderProcessor.__create_draw_rectangle_lines_ex(pos, collider))
                )
```